python2/httplib2/__init__.py

```diff
@@ -1740,9 +1756,106 @@ def clear_credentials(self):
         self.credentials.clear()
         self.authorizations = []
 
+    def add_cookie(self, name, value, max_age=None, expires=None):
+        """Set new cookie or update the existing cookie value.
+
+        max_age:
+            Number of seconds until the cookie expires
+        expires:
+            The maximum lifetime of the cookie as an HTTP-date timestamp.
+            Format:
+                <day-name>, <day> <month> <year> <hour>:<minute>:<second> GMT
+            Example: Wed, 21 Oct 2015 07:28:00 GMT
+
+        max_age has higher priority than expires.
+        """
+        if max_age:
+            expire_time = time.time() + int(max_age)
+        elif expires:
+            expire_time = calendar.timegm(email.Utils.parsedate_tz(expires))
+        else:
+            expire_time = None
+
+        self.cookies[name] = (value, expire_time)
+
+    def add_http_cookie(self, http_cookie):
+        """Set cookie with HTTP-style header(Set-Cookie:) format string.
+
+        http_cookie example:
+            sid=vBj3zbrmgynsZ8ggVpmSIOc9bXAueT3z; HttpOnly;
+
+        It may contain multiple cookies separated by comma(RFC 2109).
+
+        Example:
+            BD_NOT_HTTPS=1; path=/; Max-Age=300, BIDUPSID=D2504FE922735DE;
+            expires=Thu, 31-Dec-37 23:55:55 GMT; max-age=2147483647; path=/;
+            domain=.example.com, PSTM=1595329370
+
+        The 'expires' also contains comma. Be careful!
+        """
+        if not http_cookie:
+            return
+
+        for cookie_str in re.split(
+                r'(?:,)\s*(?=[^;, ]+?=[^;, ]+)', http_cookie):
+            attribute_list = cookie_str.split(';')
+            name, value = attribute_list[0].split('=')
+
+            max_age = expires = None
+            for attr_str in attribute_list[1:]:
+                attr = attr_str.split('=', 1)
+                if len(attr) > 1:
+                    attr_name, attr_value = attr
+                    attr_name = attr_name.strip().lower()
+                    if attr_name == 'max-age':
+                        max_age = attr_value
+                    elif attr_name == 'expires':
+                        expires = attr_value
+
+            self.add_cookie(name, value, max_age, expires)
+
+    def remove_cookie(self, name):
+        """Remove a cookie."""
+        if name in self.cookies:
+            del self.cookies[name]
```

Comment:
```suggestion
        self.cookies.pop(name, None)
```
`pop` is shorter, faster and atomic. Docstring just repeats method name and doesn't help to understand what's going on.